lib/matplotlib/backend_bases.py

```diff
@@ -3642,6 +3655,74 @@ def trigger_tool(self, name):
         """
         self.navigation.tool_trigger_event(name, sender=self)
 
+    def get_group_count(self, name):
+        """Gets useful indicies for a group.
+
+        Parameters
+        ----------
+        name : string
+            The name of the group.
+
+        Returns
+        -------
+        (start_index, group_pos) : where:
+          - `start_index, the index at the first item in the toolbar.
+          - `group_pos`, the index of this group within the toolbar.
+        """
+        index = 0
+        group_count = 0
+        for group in self._groups:
+            if group == name:
+                break
+            index += len(self._groups[group])
+            group_count += 1
+        return index, group_count
+
+    def get_positions(self, name, group, position=-1):
+        """Adds the tool to the group
+
+        Parameters
+        ----------
+        name : string
+            The name of the tool.
+        group : string
+            The name of the group.
+        position : int
+            The position within the group, if -1, it goes at the end.
+
+        Returns
+        -------
+        (position, insert_index) : where:
+
+          - `position` a validated version of the input argument, i.e. > 0.
+          - `insert index` the position used to insert into the toolbar.
+        """
+
+        if position < 0:
+            position += len(self._groups.get(group, [])) + 1
+            if position < 0:
+                position = 0
+
+        return position, self.get_group_count(group)[0] + position
+
+    def add_tool_to_group(self, tool, group, position=-1):
```

Comment:
The problem with the overall position, is that you have to decide if you include or not the `None` group,